### PR TITLE
Add onPhotoCarouselSlide prop

### PR DESCRIPTION
**Pull request by justinrex**
[Card](https://rentpath.leankit.com/card/674996770)

Add ability to pass in a function to be called when carousel is slid

### CODE REVIEW

File: packages/react-ui-ag/src/Listings/Listing.js [hunk 1]
```diff
@@ -135,13 +140,7 @@ export default class Listing extends Component {
...
   renderCtaButton(props, key) {
     const { theme, listing } = this.props
-    const {
-      className,
-      onClick,
-      valueLocation,
-      children,
-      ...rest
-    } = props
+    const { className, onClick, valueLocation, children, ...rest } = props
```
Comment (tadjohnston) [inline packages/react-ui-ag/src/Listings/Listing.js]: I personally thought these were easier to read on newlines, but to each their own i guess.

Comment (justinrex) [reply packages/react-ui-ag/src/Listings/Listing.js]: I just have prettier format things on save. I can undo if you'd like. I don't have a preference.

Comment (LinnJS) [reply packages/react-ui-ag/src/Listings/Listing.js]: I think if you would of added one more item it would have broken it out onto their own lines lol

Comment (justinrex) [reply packages/react-ui-ag/src/Listings/Listing.js]: I've seen prettier in some strange case put all but one prop on the same line, that one I really disagree with.

Comment (tadjohnston) [reply packages/react-ui-ag/src/Listings/Listing.js]: Well, we have an eslint requirement in ag.js if it's 4 or more props, put them on new lines.  We will most likely be moving to prettier anyway, so I'm sure we can discuss it then.  @LinnJS I wonder if this has to do with max line length for breaking them into new lines?

**Pull request merged**